refactoring is done

kets911 opened the pull request:
Created class hierarchy Figure -> MoveFigure -> Tractor.
Put work with moving into Orientation.
Added check if figure coordinates are less then 0.

Commit:
Created class hierarchy Figure -> MoveFigure -> Tractor.
Put work with moving into Orientation.
Added check if figure coordinates are less then 0.

ryazanovvv commented on refactoring/src/main/java/com/andersen/test/refactor/beans/Tractor.java:
```diff
@@ -0,0 +1,62 @@
+
+    @Override
+    public void move(String command) {
+        if (Constants.FORWARD_COMMAND.equals(command)) {
```
Этот if-else тоже не очень красив, количество команд может увеличиваться очень быстро, поддерживать этот if-else будет сложно

ryazanovvv commented on refactoring/src/main/java/com/andersen/test/refactor/Constants.java:
```diff
@@ -0,0 +1,7 @@
+package com.andersen.test.refactor;
+
+public class Constants {
```
Смешалось все - команды, сообщения. Предположим когда у нас количество возможных ошибок будет идти на сотни, а количество команд на тысячи - не удобно будет здесь искать нужную нам константу

ryazanovvv commented on refactoring/src/main/java/com/andersen/test/refactor/beans/Figure.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Figure {
+    protected Position position;
+    protected int[] field;
```
Не все фигуры будут двигаться по полю, предположим поезда - по железной дороге, самолеты в небе, подводные лодки - под водой. А если у нас будет какая-то составная фигура - например танк, он может двигаться и поворачивать сам, но у него еще есть башня, которая может только поворачиваться, но ее позиция меняется в зависимости от позиции танка

ryazanovvv commented on refactoring/src/main/java/com/andersen/test/refactor/beans/Figure.java:
```diff
@@ -0,0 +1,30 @@
+package com.andersen.test.refactor.beans;
+
+public class Figure {
+    protected Position position;
```
У фигур может не обязательно быть двухмерная система координат

ryazanovvv commented on refactoring/src/main/java/com/andersen/test/refactor/beans/Tractor.java:
```diff
@@ -0,0 +1,62 @@
+
+public class Tractor extends MoveFigure {
+
+    public static final String CURRENT_POSITION = "Current position: ";
```
Position вынесли в MoveFigure, а константу оставили в наследнике

Commit:
Added Place as abstraction of field.
Changed Position to interface, and created FlatnessPosition.
Realised pattern command for figure move.
Separated exception and other constants.
